### PR TITLE
Add support for Python 3.6

### PR DESCRIPTION
**Pull request by thevahidal**
Hi @hsaunders1904,
I just found your repo and wanted to thank you for creating it, it's amazing!

In my company, we have some legacy projects which still use Python 3.6, and I needed to add support for it in `pyautoenv`.
Here are the changes I made, please review them and let me know if you agree with supporting 3.6.

Thanks again!

### CODE REVIEW

**Comment by hsaunders1904** [comment]
Hi @thevahidal, thanks for your contribution, I'm glad you've found `pyautoenv` useful. I'm happy to accept your changes, however the test infrastructure currently uses poetry, which only supports Python 3.8 and above. I'll have to have a little think about how to update the test infrastructure, then I'll get back to you.

**Comment by thevahidal** [comment]
Great, let me know if I can help with anything!

File: tests/test_venv.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def test_deactivate_and_activate_switching_to_poetry(
         stdout = StringIO()
         activate_venv(self.VENV_DIR)
         # create a poetry venv to switch into
-        poetry_env = Path("poetry_proj-X-py3.8")
+        poetry_env = Path("poetry_proj-X-py3.6")
```
Comment (hsaunders1904) [inline tests/test_venv.py:131]:
```suggestion
        poetry_env = Path("poetry_proj-X-py3.8")
```

File: tests/test_poetry.py [hunk 2]
```diff
@@ -179,11 +179,11 @@ def test_poetry_cache_dir_env_var_used_if_set_and_dir_exists(self, fs):
         new_venv_dir = new_poetry_cache_dir / "virtualenvs"
         if os.name == "nt":
             new_activator = (
-                new_venv_dir / "pyproj2-lbvqfyck-py3.8" / self.activator
+                new_venv_dir / "pyproj2-lbvqfyck-py3.6" / self.activator
             )
         else:
             new_activator = (
-                new_venv_dir / "pyproj2-NKNCcI25-py3.8" / self.activator
+                new_venv_dir / "pyproj2-NKNCcI25-py3.6" / self.activator
```
Comment (hsaunders1904) [inline tests/test_poetry.py:186]:
```suggestion
                new_venv_dir / "pyproj2-NKNCcI25-py3.8" / self.activator
```

File: tests/test_poetry.py [hunk 3]
```diff
@@ -179,11 +179,11 @@ def test_poetry_cache_dir_env_var_used_if_set_and_dir_exists(self, fs):
         new_venv_dir = new_poetry_cache_dir / "virtualenvs"
         if os.name == "nt":
             new_activator = (
-                new_venv_dir / "pyproj2-lbvqfyck-py3.8" / self.activator
+                new_venv_dir / "pyproj2-lbvqfyck-py3.6" / self.activator
```
Comment (hsaunders1904) [inline tests/test_poetry.py:182]:
```suggestion
                new_venv_dir / "pyproj2-lbvqfyck-py3.8" / self.activator
```

File: tests/test_poetry.py [hunk 4]
```diff
@@ -148,9 +148,9 @@ def test_deactivate_and_activate_switching_to_new_poetry_env(self, fs):
         activate_venv(self.venv_dir)
         fs = make_poetry_project(fs, "pyproj2", Path("pyproj2"))
         if os.name == "nt":
-            new_venv = self.poetry_cache / "pyproj2-lbvqfyck-py3.8"
+            new_venv = self.poetry_cache / "pyproj2-lbvqfyck-py3.6"
         else:
-            new_venv = self.poetry_cache / "pyproj2-NKNCcI25-py3.8"
+            new_venv = self.poetry_cache / "pyproj2-NKNCcI25-py3.5"
```
Comment (hsaunders1904) [inline tests/test_poetry.py:153]:
```suggestion
            new_venv = self.poetry_cache / "pyproj2-NKNCcI25-py3.8"
```

File: tests/test_poetry.py [hunk 5]
```diff
@@ -148,9 +148,9 @@ def test_deactivate_and_activate_switching_to_new_poetry_env(self, fs):
         activate_venv(self.venv_dir)
         fs = make_poetry_project(fs, "pyproj2", Path("pyproj2"))
         if os.name == "nt":
-            new_venv = self.poetry_cache / "pyproj2-lbvqfyck-py3.8"
+            new_venv = self.poetry_cache / "pyproj2-lbvqfyck-py3.6"
```
Comment (hsaunders1904) [inline tests/test_poetry.py:151]:
```suggestion
            new_venv = self.poetry_cache / "pyproj2-lbvqfyck-py3.8"
```

File: pyautoenv.py [hunk 6]
```diff
@@ -26,14 +26,13 @@
...
-from __future__ import annotations
 
 import os
 import sys
 from functools import lru_cache
-from typing import TextIO
+from typing import List, TextIO, Union
 
-__version__ = "0.6.0"
+__version__ = "0.7.0"
```
Comment (hsaunders1904) [inline pyautoenv.py:35]:
```suggestion
__version__ = "0.6.1"
```

File: pyproject.toml [hunk 7]
```diff
@@ -4,13 +4,13 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
 name = "pyautoenv"
-version = "0.6.0"
+version = "0.7.0"
```
Comment (hsaunders1904) [inline pyproject.toml:7]:
```suggestion
version = "0.6.1"
```

File: pyautoenv.py [hunk 8]
```diff
@@ -272,14 +281,15 @@ def macos_poetry_cache_dir() -> str:
     )
 
 
-def windows_poetry_cache_dir() -> str | None:
+def windows_poetry_cache_dir() -> Union[str, None]:
     """Return the poetry cache directory for Windows."""
-    if not (app_data := os.environ.get("LOCALAPPDATA", None)):
+    app_data = os.environ.get("LOCALAPPDATA", None)
+    if not (app_data):
```
Comment (hsaunders1904) [inline pyautoenv.py:287]:
```suggestion
    if not app_data:
```

**Review by hsaunders1904 — CHANGES_REQUESTED** [review]
Hey, thanks again for your contribution, sorry for taking so long to get back to you, I haven't had a huge amount of time over the last few weeks.

I've been struggling with this a bit. Python 3.6 is very old now and is not supported by poetry or ruff; two things I'd rather not drop.

That said, I'm happy to accept your changes to the main script (with a few tweaks), so it should run with 3.6. I'll try not to introduce any changes that break things for you, but I won't be "officially" supporting 3.6.

**Comment by thevahidal** [comment]
Yeah, I totally understand and respect your decision. Thanks though that you're willing to accept these changes!

**Comment by thevahidal** [comment]
I did a new push with all the changes you asked.

File: pyautoenv.py [hunk 9]
```diff
@@ -203,29 +208,32 @@ def has_poetry_env(directory: str) -> bool:
...
     This can be found via the poetry CLI using
     ``poetry env list --full-path``, but it's painfully slow.
     """
-    if (cache_dir := poetry_cache_dir()) is None:
+    cache_dir = poetry_cache_dir()
+    if cache_dir is None:
         return []
-    if (env_name := poetry_env_name(directory)) is None:
+    env_name = poetry_env_name(directory)
+    if (env_name) is None:
```
Comment (hsaunders1904) [inline pyautoenv.py:236]: Just missed these braces here, then I think this is good to go 👍 

```suggestion
    if env_name is None:
```

**Review by hsaunders1904 — CHANGES_REQUESTED** [review]

File: pyautoenv.py (hunk 9 above)
Comment (thevahidal) [reply pyautoenv.py]: It's fixed

**Review by hsaunders1904 — APPROVED** [review]
Thanks again, @thevahidal

**Comment by thevahidal** [comment]
🙏

**Pull request merged**